src/aleph/http/client.clj

```diff
@@ -110,34 +110,32 @@
           nil))
       (no-url req))))
 
-;; xxx(errors-handling): to put exception into response-stream
-;; might not be the correct action but it would work in most cases
-;; if (and only if) we close channel right after
-;; we should somehow find a way to deal with the situation when 
-;; exception happens while realizing async body from the response
-;; (meaning that the object from response-stream is arelady taken
-;; by the user and putting a new one would simply mean we're overriding
-;; response to the next request over the same connection)
-(defn exception-handler [ctx ex response-stream]
-  (cond
-    ;; could happens when io.netty.handler.codec.http.HttpObjectAggregator
-    ;; is part of the pipeline
-    (instance? TooLongFrameException ex)
+(defn handle-exception [^ChannelHandlerContext ctx ^Throwable ex response-stream error-logger]
+  ;; Typical non-IO exception would be TooLongFrameException, SSLHandshakeError etc
+  (when-not (not (instance? IOException ex))
+    (try
+      (error-logger ex)
+      (catch Throwable e
+        (log/warn "error in error logger" e)))
```

Comment:
Do you ignore stacktrace for the exception on purpose? What I mean is that `(log/warn e "error in error logger")` would print both the exception message and its trace, but your code will only print the message itself.

Reply:
I honestly don't know if the entire trace is necessary here. If it feels like we need to log it when logging fails... I would be totally down to switch implementation.